src/app/_common/data/adData.js

```diff
@@ -0,0 +1,243 @@
+            extractedEmails.forEach((it) => {
+                if (isValidEmail(it) && !mailtoInString(preprocessedAd, it)) {
+                    preprocessedAd = preprocessedAd.replace(it, `<a rel="nofollow" href="mailto:${it}">${it}</a>`);
+                }
+            });
+            processedAdText = preprocessedAd;
+        } catch (err) {
+            processedAdText = adText;
+        }
+    }
+    // TODO: double check that sanitizing doesn't remove <section> (see amplitude.js for usage)
+    return DOMPurify.sanitize(processedAdText);
+}
+
+function getEmail(email) {
+    return isValidEmail(email) ? email : undefined;
+}
+
+function getJobPercentage(value) {
+    const jobPercentage = getString(value);
+    if (!jobPercentage) {
+        return undefined;
+    }
+
+    return jobPercentage + jobPercentage.endsWith("%") ? "" : "  %";
+}
+
+function getWorktime(value) {
+    const worktime = getString(value);
+    if (!worktime) {
+        return undefined;
+    }
+
+    // We need this check in case of old workhour/-day property values, formatted like 'Opt1 Opt2'
+    const items = [];
+    try {
+        const jsonArray = JSON.parse(worktime);
+
+        for (let i = 0; i < jsonArray.length; i += 1) {
+            items.push(jsonArray[i]);
+        }
+    } catch (e) {
+        items.push(worktime);
+    }
+    return items.join(", ");
+}
+
+function getContactList(contactList) {
+    if (!contactList) {
+        return undefined;
+    }
+
+    return contactList.map((contact) => {
+        return {
+            name: getString(contact.name),
+            title: getString(contact.title),
+            phone: getString(contact.phone),
+            email: getEmail(contact.email),
+        };
+    });
+}
+
+/**
+ *  --------------------------- Common Functions ---------------------------
+ */
+function getString(value) {
+    if (value && typeof value === "string") {
+        return value;
+    }
+    return undefined;
+}
+
+function getArray(arrayData) {
+    return Array.isArray(arrayData) ? arrayData : undefined;
+}
+
+function getUrl(url) {
+    try {
+        const urlIsValid = new URL(url).protocol.startsWith("http");
+        if (urlIsValid) {
+            return url;
+        }
+    } catch (e) {
+        return undefined;
+    }
+}
+
+function getLocationListData(locationList) {
+    if (!locationList) {
+        return [];
+    }
+    return locationList.map((location) => {
+        return {
+            address: getString(location.address),
+            city: getString(location.city),
+            postalCode: getString(location.postalCode),
+            municipal: getString(location.municipal),
+            country: getString(location.country),
+        };
+    });
+}
+
+function getJobPostingFormat(adText) {
+    if (
+        adText &&
+        adText.includes('<section id="arb-serEtter">') &&
+        adText.includes('<section id="arb-arbeidsoppgaver">') &&
+        adText.includes('<section id="arb-tilbyr">')
+    ) {
```

Comment:
Tipper denne kommer fra et annet sted, men er det ikke riktig å bruke `||` i stedet for `&&`, i tilfelle man f.eks ikke har fylt ut tilbyr-sectionen?
```
 adText &&
        (adText.includes('<section id="arb-serEtter">') ||
        adText.includes('<section id="arb-arbeidsoppgaver">') ||
        adText.includes('<section id="arb-tilbyr">'))
```

Reply:
Stillingsregistrering krever at alle tre må fylles inn. Bare åpningstekst er valgfritt. Så det spørs hvor strenge vi vil være. Dette virker som har fungert hittil i amplitude da.

Reply:
Tja er ikke helt uenig men teknisk sett så ble den lagt opp slik pga stillingsregistrering faktisk krever alle 3. Bare åpningstekst er valgfritt.